src/display-x11.cc

```diff
@@ -423,16 +423,30 @@ bool display_output_x11::main_loop_wait(double t) {
   return true;
 }
 
-#define EV_HANDLER(name)                                                     \
-  bool _conky_ev_handle_##name(conky::display_output_x11 *surface,           \
-                               Display *display, XEvent &ev, bool *consumed, \
-                               void **cookie)
-#define NOOP_EV_HANDLER(name) \
-  EV_HANDLER(name) { return false; }
+enum x_event_handler {
+  XINPUT_MOTION,
+  MOUSE_INPUT,
+  PROPERTY_NOTIFY,
+
+  EXPOSE,
+  REPARENT,
+  CONFIGURE,
+  BORDER_CROSSING,
+  DAMAGE,
+};
+
+template <x_event_handler handler>
+bool handle_event(conky::display_output_x11 *surface, Display *display,
+                  XEvent &ev, bool *consumed, void **cookie) {
+  return false;
+}
```

Comment:
Replaced `EV_HANDLER` macro with a templated function.

Nice perk with this is that there's no need for explicit `NOOP_EV_HANDLER`.